### PR TITLE
New Recipe: COSMA v2.2.0

### PR DESCRIPTION
**Pull request by haampie**
I could not get MPI to work on i686-linux-gnu-gcc (GCC) 5.2.0, will make an issue later. Instead compiled with GCC 7.

Somehow cmake did not find OpenMP on FreeBSD, maybe the mpi wrapper messes it up, have to check it later!

### CODE REVIEW

**Comment by haampie** [comment]
> Error when installing package MPICH_jll
:thinking: 

And more OpenMP issues, on macos as well.

File: C/COSMA/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    -DCOSMA_WITH_BENCHMARKS=OFF \
+    -DCOSMA_WITH_INSTALL=ON \
+    -DCOSMA_WITH_PROFILING=OFF \
+    -DBUILD_SHARED_LIBS=ON \
+    -DCOSMA_BLAS=OPENBLAS \
+    -DCOSMA_SCALAPACK=OFF \
+    -DMPI_CXX_COMPILER=$bindir/mpicxx \
+    -DMPI_C_COMPILER=$bindir/mpicc \
+    -DOPENBLAS_LIBRARIES=$BLAS_LAPACK_LIB \
+    -DOPENBLAS_INCLUDE_DIR=$libdir/../include
```
Comment (giordano) [inline C/COSMA/build_tarballs.jl:41]:
```suggestion
    -DOPENBLAS_INCLUDE_DIR="${includedir}"
```

Comment (haampie) [reply C/COSMA/build_tarballs.jl]:
```# echo "$includedir"

```
?

Comment (giordano) [reply C/COSMA/build_tarballs.jl]: You're using an old version of BinaryBuilder: https://github.com/JuliaPackaging/BinaryBuilder.jl/pull/765 :wink:

**Comment by haampie** [comment]
`_cblas_cgemm` linking issues. I'll take a look later

**Comment by haampie** [comment]
YES!

File: C/COSMA/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    -DOPENBLAS_INCLUDE_DIR=$includedir \
+    $OPENMP_CMAKE_FLAGS
+
+make -j$(nproc)
+make install
+
+# Manually copy license
+LICENSE_DIR="${prefix}/share/licenses/COSMA"
+mkdir -p "$LICENSE_DIR"
+cp ../cosma/LICENSE "$LICENSE_DIR"
```
Comment (giordano) [inline C/COSMA/build_tarballs.jl:65]: We have a single command to do that: https://juliapackaging.github.io/BinaryBuilder.jl/dev/build_tips/#Installing-the-license-file-1
```suggestion
install_license ../cosma/LICENSE
```

File: C/COSMA/bundled/patches/openblas.patch [hunk 3]
```diff
@@ -0,0 +1,76 @@
+diff --git before/cosma/src/cosma/blas.cpp after/cosma/src/cosma/blas.cpp
```
Comment (giordano) [inline C/COSMA/bundled/patches/openblas.patch:1]: Maybe I'm missing something, but isn't it easier to just rename the symbols with macros, like https://github.com/JuliaPackaging/Yggdrasil/blob/f11bf95fb657f9b7d7bc85228b9aea63f8bb278c/A/armadillo/build_tarballs.jl#L37-L39 (and here they would be much fewer symbols)?

Comment (haampie) [reply C/COSMA/bundled/patches/openblas.patch:1]: Hm yeah, that would probably be easier.

Comment (haampie) [reply C/COSMA/bundled/patches/openblas.patch]: Thanks!

**Comment by giordano** [comment]
Awesome, thanks, I can finally get rid of [my failed attempt](https://github.com/giordano/Yggdrasil/commit/7f1d23de0a59ea1e6111332644d40f0a54371db6) to build an old version of COSMA!

**Pull request merged**